modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -763,29 +763,68 @@ class CV_EXPORTS_W QRCodeEncoder {
 
 };
 
-class CV_EXPORTS_W QRCodeDetector
-{
-public:
-    CV_WRAP QRCodeDetector();
-    ~QRCodeDetector();
+struct CV_EXPORTS_W_SIMPLE QRDetectorParameters {
 
-    /** @brief sets the epsilon used during the horizontal scan of QR code stop marker detection.
-     @param epsX Epsilon neighborhood, which allows you to determine the horizontal pattern
-     of the scheme 1:1:3:1:1 according to QR code standard.
+    /** @brief Methods for detecting QR codes: CONTOURS_BASED, ARUCO_BASED */
+    enum Method {
+        CONTOURS_BASED, ///< QR detect method with detect finder pattern as contours
+        ARUCO_BASED ///< QR detect method, with detect finder pattern as ArUco marker
+        // TODO: add WECHAT method
+    };
```

Comment:
We should not have mess of parameters in a single entity.
We should explode that class. Each implementation has own parameters set.

Or event drop these "Parameters" and keep using set/get for each field.
See dnn::Model (with tests) for the reference.

Reply:
I haven't done it yet. I will do it.

Reply:
done, thx for review =)